### PR TITLE
web: Enable status messages when public code enabled

### PR DESCRIPTION
**Pull request by ryanslade**

### CODE REVIEW

**Comment by ryanslade** [comment]
Feel free to merge this on my behalf as I need to step away for a couple of hours

File: client/web/src/nav/NavLinks.tsx [hunk 1]
```diff
@@ -158,9 +158,11 @@ export const NavLinks: React.FunctionComponent<Props> = props => {
                     </li>
                 ))}
             {/* show status messages if authenticated user is admin or opted-in with a user tag  */}
-            {(authenticatedUser?.siteAdmin || authenticatedUser?.tags?.includes('AllowUserExternalServicePublic')) && (
+            {(window.context.externalServicesUserModeEnabled ||
+                authenticatedUser?.siteAdmin ||
+                authenticatedUser?.tags?.includes('AllowUserExternalServicePublic')) && (
                 <li className="nav-item">
-                    <StatusMessagesNavItem isSiteAdmin={authenticatedUser.siteAdmin} history={history} />
+                    <StatusMessagesNavItem isSiteAdmin={authenticatedUser?.siteAdmin || false} history={history} />
```
Comment (eseliger) [inline client/web/src/nav/NavLinks.tsx:165]: currently this also appears for unauthenticated users, is this desired?

Comment (ryanslade) [reply client/web/src/nav/NavLinks.tsx:165]: Good catch, we only want to show it for authenticated users

Comment (artemruts) [reply client/web/src/nav/NavLinks.tsx:165]: I have these lines on my wip notifications branch:

```js
  {/* show status messages if authenticated user opted-in with a user tag  */}
  {authenticatedUser?.tags?.includes('AllowUserExternalServicePublic') && ( 
      <li className="nav-item"> 0 
          <StatusMessagesNavItem
              isSiteAdmin={authenticatedUser.siteAdmin}
              history={history}
              userCreatedAt={authenticatedUser.createdAt}
              userID={authenticatedUser.id}
          />
      </li>
  )}
 ```
 
 Is this what we want?

File: client/web/src/nav/NavLinks.tsx [hunk 2]
```diff
@@ -158,9 +158,11 @@ export const NavLinks: React.FunctionComponent<Props> = props => {
                     </li>
                 ))}
             {/* show status messages if authenticated user is admin or opted-in with a user tag  */}
-            {(authenticatedUser?.siteAdmin || authenticatedUser?.tags?.includes('AllowUserExternalServicePublic')) && (
+            {(window.context.externalServicesUserModeEnabled ||
+                authenticatedUser?.siteAdmin ||
+                authenticatedUser?.tags?.includes('AllowUserExternalServicePublic')) && (
```
Comment (eseliger) [inline client/web/src/nav/NavLinks.tsx:163]: I think just always checking an authenticated user exists first should help :)
```suggestion
            {(authenticatedUser && (window.context.externalServicesUserModeEnabled ||
                authenticatedUser.siteAdmin ||
                authenticatedUser.tags?.includes('AllowUserExternalServicePublic'))) && (
```

Comment (ryanslade) [reply client/web/src/nav/NavLinks.tsx]: Ah, didn't see your messages. I just pushed something similar to @eseliger's solution

**Review by eseliger — APPROVED** [review]

**Comment by eseliger** [comment]
@ryanslade I saw you activated automerge but CI failed, just making sure that you noticed

**Pull request merged**